python/lsst/meas/astrom/astrometry_net.cc

```diff
@@ -0,0 +1,598 @@
+// -*- lsst-C++ -*-
+
+// Astrometry.net include files...
+extern "C" {
+#include "astrometry/solver.h"
+#include "astrometry/index.h"
+#include "astrometry/multiindex.h"
+#include "astrometry/starkd.h"
+#include "astrometry/fitsioutils.h"
+#include "astrometry/fitstable.h"
+#include "astrometry/log.h"
+#include "astrometry/tic.h"
+#include "astrometry/healpix.h"
+
+#undef ATTRIB_FORMAT
+#undef FALSE
+#undef TRUE
+
+#undef logdebug
+#undef debug
+}
+
+#include <memory>
+#include <set>
+#include <vector>
+#include <string>
+#include <sstream>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include "boost/format.hpp"
+
+#include "lsst/utils/pybind11.h"
+#include "lsst/meas/astrom/detail/utils.h"
+#include "lsst/base.h"
+#include "lsst/log/Log.h"
+#include "lsst/daf/persistence.h"
+#include "lsst/daf/base/Persistable.h"
+#include "lsst/daf/base/PropertyList.h"
+#include "lsst/pex/exceptions.h"
+#include "lsst/afw/table.h"
+#include "lsst/afw/image/Wcs.h"
+#include "lsst/afw/image/TanWcs.h"
+#include "lsst/afw/geom.h"
+#include "lsst/afw/coord.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace astrom {
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("meas.astrom.astrometry_net");
+
+struct multiindex_deleter {
+    void operator()(multiindex_t* m) {
+        multiindex_free(m);
+    }
+};
+
+struct solver_deleter {
+    void operator()(solver_t* m) {
+        solver_free(m);
+    }
+};
+
+struct timer_baton {
+    solver_t* s;
+    double timelimit;
+};
+
+static time_t timer_callback(void* baton) {
+    struct timer_baton* tt = static_cast<struct timer_baton*>(baton);
+    solver_t* solver = tt->s;
+    if (solver->timeused > tt->timelimit)
+        solver->quit_now = 1;
+    return 1;
```

Comment:
I am not very knowledgeable about C timer callbacks. However, it appears to me that this function returns 1 whether the time has elapsed or not, and that both conditions are normal. I don't see any code that handles errors.

Note that this is an exact copy of the code that was in `astrometry_net.i`. However, if it needs correcting I'd like to understand better what it needs.